rust/cardano-blockchain-types/src/point.rs

```diff
@@ -0,0 +1,570 @@
+        match other {
+            Some(cmp_hash) => {
+                match self.0 {
+                    pallas::network::miniprotocols::Point::Specific(_, ref hash) => {
+                        **hash == **cmp_hash
+                    },
+                    pallas::network::miniprotocols::Point::Origin => false,
+                }
+            },
+            None => {
+                match self.0 {
+                    pallas::network::miniprotocols::Point::Specific(_, ref hash) => hash.is_empty(),
+                    pallas::network::miniprotocols::Point::Origin => true,
+                }
+            },
+        }
+    }
+}
+
+impl Display for Point {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        if *self == Self::ORIGIN {
+            return write!(f, "Point @ Origin");
+        } else if *self == Self::TIP {
+            return write!(f, "Point @ Tip");
+        } else if *self == Self::UNKNOWN {
+            return write!(f, "Point @ Unknown");
+        }
+
+        let slot = self.slot_or_default();
+        let hash = self.hash_or_default();
+        match hash {
+            Some(hash) => write!(f, "Point @ {slot:?}:{}", hex::encode(hash)),
+            None => write!(f, "Point @ {slot:?}"),
+        }
+    }
+}
+
+impl From<pallas::network::miniprotocols::Point> for Point {
+    fn from(point: pallas::network::miniprotocols::Point) -> Self {
+        Self(point)
+    }
+}
+
+impl From<Point> for pallas::network::miniprotocols::Point {
+    fn from(point: Point) -> pallas::network::miniprotocols::Point {
+        point.0
+    }
+}
+
+impl PartialOrd for Point {
+    /// Implements a partial ordering based on the slot number
+    /// of two `Point` instances. It only checks the slot number for ordering.
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Point {
+    /// Implements a total ordering based on the slot number
+    /// of two `Point` instances. It only checks the slot number for ordering.
+    fn cmp(&self, other: &Self) -> Ordering {
+        cmp_point(&self.0, &other.0)
+    }
+}
+
+impl PartialEq<u64> for Point {
+    fn eq(&self, other: &u64) -> bool {
+        self.0.slot_or_default() == *other
+    }
+}
+
+impl PartialOrd<u64> for Point {
+    /// Allows to compare a `Point` against a `u64` (Just the Immutable File Number).
+    ///
+    /// Equality ONLY checks the Immutable File Number, not the path.
+    /// This is because the Filename is already the Immutable File Number.
+    fn partial_cmp(&self, other: &u64) -> Option<Ordering> {
+        self.0.slot_or_default().partial_cmp(other)
+    }
+}
+
+impl PartialEq<Option<Point>> for Point {
+    /// Allows to compare a `SnapshotID` against `u64` (Just the Immutable File Number).
+    ///
+    /// Equality ONLY checks the Immutable File Number, not the path.
+    /// This is because the Filename is already the Immutable File Number.
```

Comment:
Am I missing something or this comment is wrong? It is the same as in the `PartialOrd<u64> for Point` implementation.

Reply:
The comment is wrong if its the same as another comment.